Dickey

ThomasDickey opened the pull request:
Updated/corrected tests for xterm 332/333.

ThomasDickey committed:
originally 2018-04-04

	* tests/da.py:
	works for me (I fixed that in January 2015 from report by Nachman)

	* tests/decrqss.py, tests/decera.py, tests/decsera.py, tests/decic.py, tests/decdc.py, tests/xterm_winops.py:
	works for me

ThomasDickey committed:
originally 2018-04-05

	* tests/decset.py: works for me

	* tests/decscl.py: works for me (DECSCL except for the "expected" RIS)

	* tests/deccra.py: works for me - but his expected value was wrong

ThomasDickey committed:
originally 2018-04-06

	* tests/decset.py, tests/rep.py: fixed:332

ThomasDickey committed:
originally 2018-04-08

	* tests/xterm_winops.py:
	try to restore screen between maximizing (would need a composite test too)

	* esccmd.py: another code

	* tests/xterm_winops.py:
	try adding timing fixes (doesn't seem to help, but doesn't hurt)

	* tests/xterm_winops.py:
	hmm - no progress, but I see that some "Passes" are due to starting from
	a maximized state.

	* tests/xterm_winops.py: still 3 (problem with iconify/deiconfy?)

	* tests/xterm_winops.py: 3...

	* tests/xterm_winops.py: 4...

	* tests/xterm_winops.py: 5...

	* tests/xterm_winops.py: 6...

	* tests/xterm_winops.py:
	partial repair for fvwm - down to 7 failures (no change for XQuartz)

	* escutil.py: add functions for new xterm controls

	* esccmd.py: add codes for new xterm controls

	* tests/xterm_winops.py:
	partly improved: I found that XQuartz is ignoring the maximize horz/vert
	calls though it honors fullscreen.  When switching to fvwm session, I
	get a bunch of regressions due to confusing pixels and chars.  Check this
	in, to focus on those...

	* tests/xterm_winops.py: another bogus test

	* tests/xterm_winops.py: 2 bogus tests fixed.

	* tests/xterm_winops.py: explain

	* tests/xterm_winops.py: fixed test

	* tests/ris.py: fix

	* tests/decset.py: corrected test

	* tests/save_restore_cursor.py: xterm #328 has fix for wraptest -

ThomasDickey committed:
originally 2018-04-09

	* tests/decdc.py: perhaps he needs a better test...

	* tests/decfi.py, tests/decbi.py: clarified

	* tests/cht.py: test was wrong - fix

	* tests/bs.py: align with xterm #332

	* tests/dl.py: fixed

	* esc.py: add BEL, so I can send it in a test to pinpoint trace

	* tests/xterm_winops.py: comment-out - some fail, will revisit

	* tests/sd.py: works now :-)

	* tests/ich.py: works for me

ThomasDickey committed:
originally 2018-04-10

	* tests/decrqm.py, esccmd.py: drop DECVSSM

	* escutil.py: *** empty log message ***

	* tests/decdsr.py: levels

	* escutil.py: fix his typo

	* tests/decscl.py, tests/decrqss.py: levels

	* esccmd.py:
	DECRQSS not found in vt320 manual (which is PDF only - no searching)

	* tests/decrqm.py, tests/vt.py, tests/vpr.py, tests/vpa.py, tests/su.py, tests/sos.py, tests/sm.py, tests/ris.py, esccmd.py, tests/s8c1t.py, tests/sd.py, tests/rm.py, tests/ri.py, tests/rep.py, tests/pm.py, tests/nel.py, tests/lf.py, tests/ind.py, tests/il.py, tests/ich.py, tests/hvp.py, tests/hpr.py, tests/hpa.py, tests/el.py, tests/ff.py, tests/ed.py, tests/ech.py, tests/dl.py, tests/decstbm.py, tests/decstr.py:
	levels

	* tests/decset.py, tests/decsera.py: levelsdecsel.py

	* tests/decsel.py, tests/decsed.py, tests/decscl.py, tests/decaln.py, tests/dcs.py, tests/apc.py, tests/dch.py, tests/cup.py, tests/cuf.py:
	levels

	* tests/cub.py, tests/cr.py, tests/cpl.py, tests/cnl.py, tests/cbt.py, tests/cht.py, tests/cha.py, tests/bs.py:
	level

	* tests/save_restore_cursor.py: check levels

	* tests/decrqm.py: filter out some of the "known bug" DECRQM

	* escutil.py, esccmd.py: better

	* escutil.py: better.

	* escutil.py: can't do DECRQCRA in a vt100...

	* tests/da.py, tests/decrqss.py: trim

	* tests/decrqss.py: better

	* tests/da.py: first cut/workable DA response

ThomasDickey committed:
originally 2018-04-11

	* tests/deccra.py, tests/decdc.py, tests/decera.py, tests/decic.py, tests/decscl.py, tests/decset.py, tests/rep.py:
	not-xterm

	* tests/xterm_winops.py:
	python functions need parenthese - George pointed that out, and pylint too

	* tests/da2.py: match xterm's default behavior

ThomasDickey committed:
originally 2018-04-12

	* escio.py, tests/dl.py, tests/hpa.py, tests/hvp.py, tests/ich.py, tests/sd.py, tests/ff.py, tests/ech.py, tests/ed.py, tests/el.py, tests/lf.py, tests/reset_color.py, tests/reset_special_color.py, tests/tbc.py, tests/vpa.py, tests/vpr.py, tests/decset.py, tests/decstbm.py, tests/decstr.py, tests/decscl.py, tests/decsed.py, tests/decsel.py, tests/decdsr.py, tests/decera.py, tests/decfi.py, tests/decfra.py, tests/decic.py, tests/decid.py, tests/decrqm.py, tests/decrqss.py, tests/il.py, tests/ris.py, tests/rm.py, tests/s8c1t.py, tests/sm_title.py, tests/vt.py, tests/xterm_winops.py, tests/rep.py, tests/su.py, tests/decbi.py, tests/deccra.py, tests/decdc.py, tests/dcs.py, tests/decaln.py, tests/dch.py, tests/cub.py, tests/cuf.py, tests/cup.py, tests/da2.py, tests/cr.py, tests/cha.py, tests/change_special_color.py, tests/change_dynamic_color.py, tests/change_color.py, tests/cbt.py, tests/bs.py, tests/ansirc.py, escutil.py, esclog.py, esc.py:
	pylint-fixes

ThomasDickey committed:
originally 2018-04-14

	* tests/save_restore_cursor.py: tidy

	* tests/decbi.py: use double-quotes for docstring

	* tests/decfi.py, tests/decdc.py: docstring

	* tests/decbi.py: comments vs docstring

	* tests/decset.py: remove obsolete comment

	* tests/xterm_winops.py: remove redundant reset

	* tests/ris.py, tests/decset.py, tests/rep.py, tests/sd.py:
	remove comment vs xterm

	* tests/decset.py: no blank before docstring

	* tests/decdsr.py: fix a symbol conflict

ThomasDickey committed:
originally 2018-04-15

	* tests/xterm_winops.py:
	Nachman's objecting to use of char-sizes when he asked for pixels.
	Start refactoring by using an if-statement to restore his original
	version (except for a couple of cases where the naming wasn't consistent,
	including one where the name was incorrect - seen with pylint).

ThomasDickey committed:
originally 2018-04-17

	* tests/cht.py: document test_CHT_IgnoresScrollingRegion

	* tests/bs.py: explain reverse-wrap versus margins

	* tests/xterm_winops.py: fix per Nachman comment:
	- Move the call to unmaximize before the assertion to improve the odds of it happening in:
	  * test_XtermWinops_MaximizeWindow_Horizontally
	  * test_XtermWinops_MaximizeWindow_Vertically

	* escutil.py: change commented-out LogInfo to a LogDebug

	* tests/xterm_winops.py: fix per Nachman comment:
	- In test_XtermWinops_MaximizeWindow_HorizontallyAndVertically,
	move esccmd.XTERM_WINOPS(esccmd.WINOP_MAXIMIZE, esccmd.WINOP_MAXIMIZE_EXIT)
	to before the assertions. If they fail the window will be stuck maximized.

ThomasDickey committed:
originally 2018-04-18

	* tests/xterm_winops.py:
	factor out average_width() and average_height(), fix whitespace per pylint

	* tests/xterm_winops.py:
	use absolute difference in some assert's for consistency, per Nachman comment

	* tests/sd.py:
	cite relevant standard for module docstring, fix whitespace per pylint

	* tests/save_restore_cursor.py:
	add docstring to test_SaveRestoreCursor_AltVsMain and
	test_SaveRestoreCursor_Wrap, and fix whitespace per pylint

	* tests/ris.py:
	add docstring to test_RIS_ResetDECCOLM, fix whitespace per pylint

	* tests/rep.py, tests/decfi.py, tests/decdc.py:
	cite relevant standard for module docstring, fix whitespace per pylint

	* tests/decdc.py:
	quote the relevant standard for test_DECDC_IsNoOpWhenCursorBeginsOutsideScrollRegion

	* tests/decbi.py:
	cite relevant standard for test_DECBI_LeftOfMargin and test_DECBI_WholeScreenScrolls

ThomasDickey committed:
originally 2018-04-20

	* tests/xterm_winops.py:
	the remaining failed-test was due to inconsistent max_error value.

	* tests/xterm_winops.py:
	add/use delayAfterMove, delayAfterIcon, delayAfterResize, making all but
	the 2-way maximize work consistently with icewm

ThomasDickey committed:
originally 2018-04-21

	* tests/xterm_winops.py:
	use checkActualPixelSize() for the remaining xterm-resizing tests involving
	resizing by pixels

	* tests/xterm_winops.py:
	refactor test_XtermWinops_ResizePixels_BothParameters() using new function
	checkActualPixelSize()

	* escutil.py: add GetFrameSizePixels()

	* escutil.py: cache the char/screen sizes, fix whitespace per pylint

ThomasDickey committed:
originally 2018-04-22

	* tests/xterm_winops.py: factor out check_for_shrinkage()

	* tests/xterm_winops.py: factor out check_any_size

	* tests/xterm_winops.py: factor out check_actual_size_chars()

	* tests/xterm_winops.py:
	reduce delays, corrected a label on informational log

ThomasDickey committed:
originally 2018-04-23

	* tests/xterm_winops.py:
	tighten pixel error-limit to 1 cell; seems to work with quartz-wm

	* escutil.py:
	add/use can_query_shell_size(), which will allow tightening the error-limits
	for xterm's window operations

	* escutil.py, tests/xterm_winops.py:
	change level of new debug-log calls

	* tests/xterm_winops.py:
	reducing delay after iconify caused a test-failure - undo that one.

	* tests/decdsr.py, tests/deccra.py, tests/decbi.py, tests/decaln.py, tests/dch.py, tests/da.py, tests/cup.py, tests/cr.py, tests/cha.py, tests/change_special_color.py, tests/apc.py, esctypes.py, esctest.py, escio.py, esccmd.py, escargs.py:
	fix whitespace per pylint

ThomasDickey committed:
originally 2018-04-24

	* tests/sm_title.py, tests/__init__.py, tests/ed.py, tests/deccra.py, tests/vpr.py, tests/vt.py, tests/su.py, tests/sos.py, tests/nel.py, tests/pm.py, tests/ri.py, tests/rm.py, tests/s8c1t.py, tests/sm.py, tests/il.py, tests/ind.py, tests/lf.py, tests/fill_rectangle.py, tests/hpr.py, tests/hvp.py, tests/ich.py, tests/ff.py, tests/el.py, tests/ech.py, tests/dl.py, tests/decstr.py, tests/decstbm.py, tests/decset.py, tests/decsera.py, tests/decic.py, tests/decsel.py, tests/decsed.py, tests/decscl.py, tests/decrqm.py:
	fix whitespace per pylint

	* tests/xterm_winops.py:
	add documentation for the utility functions used for xterm-tests

	* README.txt:
	the section on xterm configuration options was incomplete.  Adapted from
	cfg-esctest/cfg-esctest-wide scripts.

	* README.txt:
	cite DEC standards documentation, which is used in xterm versus the
	programmer reference manuals which give insight into differences between
	terminal models but are not authoritative.  Also mention an old bug-fix.

gnachman commented on esctest/escutil.py:
```diff
@@ -1,26 +1,32 @@
+CHAR_SIZE_PIXELS = Size(0, 0)
+FRAME_SIZE_PIXELS = Size(0, 0)
+SCREEN_SIZE_PIXELS = Size(0, 0)
+CAN_QUERY_SHELL_SIZE = -1
```
I've been using SCREAMING_SNAKE_CASE for constants and gCamelCase for global variables. These are more like global variables than constants since they get assigned to, so please name them like `gCharSizePixels`, etc.

ThomasDickey replied:
ok

gnachman commented on esctest/escutil.py:
```diff
@@ -45,16 +51,109 @@ def GetWindowTitle():
+  The "screen" refers to the X display on which xterm is running.
+  Because that does not change as a result of these tests, we
+  cache a value to help with performance."""
+  global SCREEN_SIZE_PIXELS
```
iTerm2 doesn't support this since the notion of screen is ambiguous when there are multiple displays. I'm not sure if VTE does or not. How about if this returns a fixed value of 1024x768 if the terminal is not xterm?

ThomasDickey replied:
ok

gnachman commented on esctest/tests/decset.py:
```diff
@@ -436,7 +444,7 @@ def test_DECSET_OPT_ALTBUF_CURSOR(self):
     self.doAltBuftest(esccmd.OPT_ALTBUF_CURSOR, True, True, True)
 
   # xterm doesn't implement auto-wrap mode when wide characters are disabled.
```
Is this comment still true? If not, remove it; if so, why remove `@optionRejects`?

ThomasDickey replied:
I overlooked the comment :-)

gnachman commented on esctest/tests/ris.py:
```diff
@@ -49,6 +49,8 @@ def test_RIS_ResetTitleMode(self):
     AssertEQ(GetIconTitle(), "a")
 
+  @vtLevel(4)
+  @knownBug(terminal="iTerm2", reason="iTerm2 doesn't support ALTBUF.")
```
iTerm2 does support ALTBUF. Did you find a bug in it?

ThomasDickey replied:
I overlooked the change when I prepared the branch (recall that I started from the iTerm2 repo).

gnachman commented on esctest/tests/save_restore_cursor.py:
```diff
@@ -133,7 +153,10 @@ def test_SaveRestoreCursor_Wrap(self):
     esccmd.CUP(Point(GetScreenSize().width() - 1, 1))
     escio.Write("abcd")
-    AssertEQ(GetCursorPosition().y(), 2)
+    if escargs.args.expected_terminal == "xterm":
```
What's the reason for the special xterm behavior? It's nice to document things like this in a comment.

ThomasDickey replied:
That's answered already in the docstring.

gnachman replied:
> Older versions of xterm did not save/restore the last-column flag in DECRC, causing the cursor to be the second column rather than the first…

If I understand correctly, the test as written will pass for "older versions" of xterm and (presumably) fail on the latest version. Wouldn't it make more sense to test the current code?

ThomasDickey replied:
It's testing the current code.  I changed this about a year ago

https://invisible-island.net/xterm/xterm.log.html#xterm_328

* modify DECRC to save/restore xterm's last-column flag used to control wrapping behavior rather than manipulating DECAWM (report/analysis by Mattias Engdegård).

gnachman commented on esctest/tests/xterm_winops.py:
```diff
@@ -12,288 +16,599 @@
 # 9;0 - Restore maximized window
 
 class XtermWinopsTests(object):
+  def delay_after_icon(self):
```
Functions follow a naming convention of CamelCase, so this should be `DelayAfterIcon`. The only exception is tests, which much begin with `test_` to be recognized as tests.

ThomasDickey replied:
ok

gnachman commented on esctest/tests/xterm_winops.py:
```diff
@@ -12,288 +16,599 @@
-    esccmd.XTERM_WINOPS(esccmd.WINOP_RESIZE_PIXELS,
-                            original_size.height(),
-                            original_size.width())
+    if escargs.args.expected_terminal == "xterm":
```
Did you go with separate tests for xterm because iTerm2 doesn't support GetScreenSizePixels()? It would be nice to share the same tests if possible.

ThomasDickey replied:
yes - some merging would be possible, but I didn't do that, to make the differences easier to follow.

gnachman replied:
Once GetScreenSizePixels has been modified to return a constant, it should be possible to reduce the amount separate xterm code in these tests. It would be good to minimize complexity and test the same thing across different terminals.

ThomasDickey replied:
It does this if the terminal isn't xterm:

   gScreenSizePixels = Size(1024, 768)

ThomasDickey committed:
originally 2018-04-25

	* tests/xterm_winops.py, escutil.py: stylistic change per Nachman

	* tests/ris.py:
	trim comment for iTerm2 vs ALTBUF, overlooked change in master by Nachman

	* tests/decset.py:
	remove an overlooked/obsolete comment for test_DECSET_DECLRMM()

	* escutil.py:
	if the terminal is not xterm, GetScreenSizePixels() will just return a
	fixed size, suggested by Nachman

	* escutil.py: stylistic change per Nachman

gnachman commented:
I've squashed and merged the commits. I made a few small changes, especially in xterm_winops, to reduce the amount of forking. Please test that I didn't break anything on the xterm side.

Finally, thank you for taking on this huge change! I really appreciate it.

ThomasDickey commented:
thanks (will retest/etc)
